### ISSUE TITLE
how can i get the true physical depth info from the scene image?

### ISSUE THREAD

**Issue by cheng-lao** [jiaxiZeng/Temporally-Consistent-Stereo-Matching#9]
hello, thanks for your excellent work!!

i have read the `evaluate_stereo.py` script content partially, i wanted to know how to run the model on my custom data, and got the true physical depth value after getting disparity map.

first i ran the `submit_kitti.sh` to get the disparty map (named `000000_10.png` in the `kitti_15_seq_out/disp_0`),, then i used the function `disp2depth` in `core/utils/geo_utils.py` to change the disparty map into depth map.  i set the parameter `baseline = 0.54` and `fx = 7.215377e+02` .( i read the config file in the `datasets/KITTI/kitti_seq/kitti2015_testings/sequences/000000` and `P_rect_02[0][0] == 7.215377e+02`)

code as follows:
```python
import cv2
import torch
from core.utils.geo_utils import depth2disp, disp2depth
import numpy as np


disp_path = '/home/chengzi/ProgramData/Temporally-Consistent-Stereo-Matching/kitti_15_seq_out/disp_0/000000_10.png'
disp_image = cv2.imread(disp_path, cv2.IMREAD_GRAYSCALE)
disp = torch.tensor(np.array(disp_image)).cuda(0)   # H,W,C

fx = 7.215377e+02
fx = torch.tensor(fx).cuda(0)[None]

baseline = torch.tensor(0.54).float().cuda(0)[None]
depth = disp2depth(disp, baseline, fx)
depth = depth.squeeze().detach().cpu().numpy()
depth = cv2.normalize(depth, None, 0, 255, cv2.NORM_MINMAX).astype('uint8')

cv2.imwrite("./depth.jpg", depth)
```
and i got the `depth.jpg`

![Image](https://github.com/user-attachments/assets/d67ce0a7-629e-4c21-ab5b-5d6b857d4872)

but the rgb image(`datasets/KITTI/kitti_seq/kitti2015_testings/sequences/000000/image_2/000000_10.png`) is 

![Image](https://github.com/user-attachments/assets/93e138f7-9bf2-4cee-9e62-b0cdb02c3714)

apparently, these doesn't match.  (For example, the sky in the upper distance of the image did not turn white, but the center of the depth map did. Also, the road sign on the right side of the image has no outline! So this depth isn't correctly and i can’t get the true physical depth map! )

so i want to ask this question !! thanks!

**Comment by jiaxiZeng** [comment]
Thanks for point out this issue. Since the KITTI dataset lacks disparity supervision for the sky at infinite distances, most methods trained on it struggle to handle the infinite, textureless sky regions. Additionally, the errors in the outline of the road sign are caused by the blending of textureless backgrounds with foreground objects, making it difficult for the model to distinguish between them.

**Comment by cheng-lao** [comment]
> Thanks for point out this issue. Since the KITTI dataset lacks disparity supervision for the sky at infinite distances, most methods trained on it struggle to handle the infinite, textureless sky regions. Additionally, the errors in the outline of the road sign are caused by the blending of textureless backgrounds with foreground objects, making it difficult for the model to distinguish between them.

Thanks for your reply. **But I don't know how to get the real physical depth map (the actual physical meaning of the value of each pixel in the depth map is the distance in meters), what should I do?**
I found that there are some errors in my code, here is the modified code:
```python
import cv2
import torch
from core.utils.geo_utils import  disp2depth
import numpy as np


disp_path = '/home/chengzi/ProgramData/Temporally-Consistent-Stereo-Matching/kitti_15_seq_out/disp_0/000000_10.png'
disp_image = cv2.imread(disp_path,  cv2.IMREAD_UNCHANGED)  # modify here !
disp_image = np.array(disp_image).astype(np.float32) / 256
disp = torch.tensor(disp_image).cuda(0)   # H,W,C

fx = 7.215377e+02
fx = torch.tensor(fx).cuda(0)[None]

baseline = torch.tensor(0.54).float().cuda(0)[None]
depth = disp2depth(disp, baseline, fx)
depth = depth.squeeze().detach().cpu().numpy()
depth = cv2.normalize(depth, None, 0, 255, cv2.NORM_MINMAX).astype('uint8')

cv2.imwrite("./depth.jpg", depth)
```
After modifying the code the depth map looks more accurate.

![Image](https://github.com/user-attachments/assets/f314c39f-bf48-4635-b883-5b713b47ef8e)

**Comment by jiaxiZeng** [comment]
@cheng-lao Your code for converting disparity to depth is correct; the core conversion formula is:

```
depth = baseline * fx / disp
```

However, I do not recommend saving it as a uint8 grayscale image because that will result in significant precision loss. If you only need it for visualization, you might consider using a colormap (e.g., jet) for color mapping. If you intend to use it for downstream processing, you might consider saving it as a .npy file.

**Comment by cheng-lao** [comment]
@jiaxiZeng 
OK, thanks for your suggestion! :heart:
But my problem is still not solved. :joy: :disappointed_relieved:
My question is how to get the real physical depth information (the actual physical meaning of each pixel value in the depth map is the distance in meters)?

I want to know if I can only get the physical depth information through this model?
In other words, maybe I need some other information, such as IMU data or some key data, through these additional data, I can get what I want. :grey_question:

**Comment by jiaxiZeng** [comment]
@cheng-lao The depth calculate from the disparity is the the real physical depth. In addition to the rectified stereo images, you will also need the intrinsic parameters of the left camera, the baseline length, and the pose of each frame (which can be calculated from the IMU data).

**Comment by cheng-lao** [comment]
@jiaxiZeng 
OK, thanks for all your reply ! :sparkling_heart:

Wish you good luck with your internship in DJI ! :muscle:
